### PR TITLE
wispr xml inside captive-portal redirection

### PR DESCRIPTION
**Pull request by obilodeau**
Ready for review. 

NEWS

```
Enhancements
 * Better WISPR support
 * New parameter to control captive portal interception: secure_redirect
```

### CODE REVIEW

**Comment by dwlfrth** [comment]
commit 06df64f is kind of heavy to review.
been through it attentively and everything seems fine but i think someone else should take a look at it or we should put it under tests.

**Comment by obilodeau** [comment]
I re-tested the whole workflow (including passthroughs) on the iPad in the lab and it worked. I think it would be reasonable to merge this in devel today.

File: lib/pf/web/dispatcher.pm [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    # passthrough
+    # if the regex is not defined, we skip, this allow us to skip an expensive Config test
+    if (defined $CAPTIVE_PORTAL{'PASSTHROUGH_HOSTS_RE'} ) {
+        return Apache2::Const::DECLINED if (_matches_passthrough($r));
+    }
+
+    # fallback to a redirection
+    $r->uri('/perl/portal-redirect');
+    return Apache2::Const::DECLINED;
```
Comment (fdurand) [inline lib/pf/web/dispatcher.pm]: you can go directly to handler without going through a redirection.
            $r->set_handlers( PerlResponseHandler => &handler );
            return Apache2::Const::OK;

Comment (obilodeau) [inline lib/pf/web/dispatcher.pm]: I tried with both `return Apache2::Const::OK;` and `return Apache2::Const::DECLINED;` but I still get an undef `$r` once inside the handler.

Comment (obilodeau) [inline lib/pf/web/dispatcher.pm]: Ok finally nailed it, I was missing a `$r->handler('modperl');` statement.

**Pull request merged**